pfam_experiment.py

```diff
@@ -87,142 +127,171 @@ def main(_):
+
+		trained_params = copy.deepcopy(optimizer.target.params)
+
+		results, preds = pfam_evaluate(predict_fn=optimizer.target,
+	                                   test_family_accessions=lens_knn_train_family_accessions,
+	                                   title=None,
+	                                   loss_fn_kwargs=loss_fn_kwargs,
+	                                   batch_size=FLAGS.batch_size)
+
+		lens_accuracy = results['accuracy']
+		datum['lens_accuracy' + '_measurement_' + str(i)] = lens_accuracy
+
+		lens_cross_entropy = float(results['cross_entropy'])
+		datum['lens_cross_entropy' + '_measurement_' + str(i)] = lens_cross_entropy
+
+		assert(embedding_optimizer.target.params.keys()==trained_params.keys()), 'Optimizer parameters do not match!'
+		for layer in embedding_optimizer.target.params.keys():
+			embedding_optimizer.target.params[layer] = trained_params[layer]
+
+		train_knn_results_trained_lens = pfam_nearest_neighbors_classification(encoder=embedding_optimizer.target, 
+					                                                               train_family_accessions=lens_knn_train_family_accessions, 
+					                                                               test_family_accessions=lens_knn_train_family_accessions,
+					                                                               batch_size=FLAGS.batch_size,
+					                                                               train_samples=1,
+					                                                               seed=1,
+					                                                               random_state=1)[0]
+		train_knn_accuracy_trained_lens = train_knn_results_trained_lens['1-nn accuracy']
+		datum['train_knn_accuracy_trained_lens_1_knn_train_samples' + '_measurement_' + str(i)] = train_knn_accuracy_trained_lens
+
+		for knn_train_samples in knn_train_samples_:
+
+			test_knn_results_trained_lens = pfam_nearest_neighbors_classification(encoder=embedding_optimizer.target, 
+					                                                              train_family_accessions=knn_test_family_accessions, 
+					                                                              test_family_accessions=knn_test_family_accessions,
+					                                                              batch_size=FLAGS.batch_size,
+					                                                              train_samples=knn_train_samples,
+					                                                              seed=1,
+					                                                              random_state=1)[0]
+			test_knn_accuracy_trained_lens = test_knn_results_trained_lens['1-nn accuracy']
+			datum['test_knn_accuracy_trained_lens_' + str(knn_train_samples) + '_knn_train_samples' + '_measurement_' + str(i)] = test_knn_accuracy_trained_lens
+
+		assert(model.params.keys()==trained_params.keys()), 'Model and optimizer parameters do not match!'
+		predict_fn_params = trained_params[layers[-1]]
+		if trainable_encoder:
```

Comment:
see above comments. I don't think this is necessary if you make the model outside of the loop.

Reply:
yeah, model is initially created outside of the loop, but I'm not sure if I can treat the optimizer.target object as a model, I'll try this and if it works replace it

Reply:
I believe you can treat optimizer.target as a model

Reply:
yes, this worked!